browser/ui/browser_commands.cc

```diff
@@ -62,4 +70,27 @@ void OpenGuestProfile() {
   profiles::SwitchToGuestProfile(ProfileManager::CreateCallback());
 }
 
+void ToggleSpeedreader(Browser* browser) {
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+  speedreader::SpeedreaderService* service =
+      speedreader::SpeedreaderServiceFactory::GetForProfile(browser->profile());
+  if (service) {
+    service->ToggleSpeedreader();
+
+    // TODO(iefremov): We probably should do it with lesser amount of hacks.
+    BrowserView* view = static_cast<BrowserView*>(browser->window());
+    SpeedreaderButton* button =
+        static_cast<BraveToolbarView*>(view->toolbar())->speedreader_button();
+    if (button) {
+      button->Toggle();
+    }
```

Comment:
How about control button's toggle status by button itself by overriding `OnMouseRelease()`?

Reply:
The command may be executed not only by mouse, so I think it's better as is

Reply:
Yeah, accelerator could run it. I'm still thinking about how to delete this code...
Ok - how about update button's on/off state by checking `kSpeedreaderEnabled` value in `Button::Update()`?
Or monitoring that prefs in the button?

Reply:
I'm not in favor of splitting this logic... Probably we should just patch or inherit `Browser`. I've placed a TODO here, mb we can think about it in follow-ups?

Reply:
For me, most important thing here is avoiding layer violation. I'm fine both adding new method to BraveBrowserView or letting button handle its on/off status by itself.
If you want to fix this layer violation later, it's fine for me. but why not now?

Reply:
Ah, my bad, I didn't notice this is a layering violation as well. OK, I'll push a fix

Reply:
fixed